docs/adding-and-updating-views.rst

```diff
@@ -0,0 +1,189 @@
+
+2. We need to add document counts for `planning` and `tender` objects. They already exist for `award` and `contract`, and the structure is the same, so we can copy one of the existing queries and change the field names, eg:
+
+.. code-block:: sql
+
+    -- We add all of this, copying from tmp_award_documents_aggregates
+
+    drop table if exists tmp_planning_documents_aggregates;
+    create table tmp_planning_documents_aggregates
+    AS
+    select 
+        id, 
+        jsonb_object_agg(coalesce(documentType, ''), documentType_count) planning_documentType_counts
+    from
+        (select 
+            id, documentType, count(*) documentType_count
+        from
+            planning_documents_summary
+        group by
+            id, documentType
+        ) AS d
+    group by id;
+
+    create unique index tmp_planning_documents_aggregates_id on tmp_planning_documents_aggregates(id);
+
+3. Next we need to see where the intermediary :code:`tmp_` tables are used, and add our new table in there as well. Later in the file are joins which connect the aggregates to the summary table. The order of the joins is the order the columns appear in the table, so place new ones according to how you want to see the output:
+
+.. code-block:: sql
+
+    -- We add this with the rest of the joins in the file
+    left join
+        (select id, documents_count AS total_planning_documents from planning_summary) AS planning_summary using(id) 
+    left join
+        tmp_planning_documents_aggregates
+    using(id)
+
+4. Finally we drop the :code:`tmp_` table since we don't need it any more:
+
+.. code-block:: sql
+
+    drop table if exists tmp_planning_documents_aggregates;
+
+5. Now :ref:`test your changes<testing-changes>`.
+
+6. Then :ref:`add documentation<adding-docs>`. This is **required**!
+
+7. Run the tests to make sure your changes were successful and you didn't break anything else in the process by running: :code:`pytest`
+
+.. _testing-changes:
+
+Testing your changes
+--------------------
+
+Test your update by comparing it to the initial view you made when :ref:`loading data<loadingdata>`.
+
+  * Either create a new view: :code:`python ocdskingfisher-views-cli add-view 1 "Test: view with descriptions"`
+  * or refresh your existing view: :code:`python ocdskingfisher-views-cli refresh-views collection_1`
+  * Verify that the data is what you expect it to be.
+  * (If you're looking at the data in a postgres client, don't forget to refresh it.)
+
+.. _adding-docs:
+
+Adding documentation
+--------------------
+
+The tests won't pass if you don't document new fields!
+
+1. Add changes to the inline field-level documentation as follows:
+  * Edit the :code:`999-docs.sql` file to add the new fields and their descriptions as comments on columns. The comments should be in the same order as the tables.
```

Comment:
Sorry, this might just be me after lunch but I don't understand what "The comments should be in the same order as the tables." means?